### ISSUE TITLE
Unexpected truncation in flex container with BrowserAnimationsModule

### ISSUE THREAD

**Issue by altsyvanovich** [lentschi/ngx-nested-ellipsis#17]
In case if project has BrowserAnimationsModule and an observed element inside a flex container then this element can be truncated even if it shouldn't be.
It happens because when you call `this.viewContainer.clear()` inside `restoreView` method then BaseAnimationRenderer doesn't remove child immediately, it just marks child as should be deleted. Looks like that is why the code decides that there is not enough place and truncates the element.

Here is a [reproduction](https://stackblitz.com/edit/github-amrxnf?file=projects%2Fngx-nested-ellipsis%2Fsrc%2Flib%2Fdirectives%2Fnested-ellipsis.directive.ts,src%2Fapp%2Fapp.module.ts,src%2Fapp%2Fapp.component.html) - a little bit updated your demo by including BrowserAnimationsModule and adding an example to the top. As you can see `Test label` is truncated, but actually it has enough place to be fully displayed. If you remove BrowserAnimationsModule or make container non-flex then label will be fine and will not be truncated. Also you can play with container's width and find some width when the label is not truncated.

To workaround this issue we are using css styles to display just the last `nested-ellipsis-content` node:

```
nested-ellipsis-content {
    display: none;
}

nested-ellipsis-content:last-of-type {
    display: block;
}
```

**Comment by lentschi** [comment]
@altsyvanovich Sorry for not responding sooner!
I've tried several approaches to solve this issue, but none has succeeded so far, so I've now posted a SO question:
https://stackoverflow.com/questions/74923278/how-to-make-the-angular-renderer-act-as-if-animations-had-been-disabled
I hope someone can provide a solution there - if not, I'll just have to add a disclaimer that the lib is incompatible with any kind of AnimationsModule. (Oddly happens with the `NoopAnimationsModule` too.) :disappointed:

**Comment by lentschi** [comment]
Actually after I posted the SO question and commented here, I _could_ think of a solution.
But it's more of a workaround than a clean fix: https://github.com/lentschi/ngx-nested-ellipsis/commit/ba5c6a614ddd8a82dc9028dd6d82a304d39ba2d0

So I'll still wait a bit for responses to the SO question before I merge this.

**Comment by lentschi** [comment]
Since there have been no answers to my SO question, I decided to publish the workaround. It's included in `ngx-nested-ellipsis@2.1.2`.
